purchase_requisition_full_bid_order_generator/models/bids.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api
+
+
+class PurchaseRequisition(models.Model):
+    _inherit = 'purchase.requisition'
+
+    @api.multi
+    def create_purchase_order(self):
+        partner_obj = self.env['res.partner']
+        suppliers = partner_obj.search([('supplier', '=', True)])
+        supplier_obj = self.env['product.supplierinfo']
+
+        suppliers_list = []
+        for supplier in suppliers:
```

Comment:
Esto es muy poco eficiente cuando se tienen muchos productos. Lo que hay que hacer es recorrer cada línea del bid, y ver que líneas de product.supplierinfo existen para ese producto. Con la función mapped, obtienes el supplier de esas líneas, y se te queda guardado como un recordset, que actúa como un conjunto (el concepto matemático). Para la siguiente línea, vuelves a hacer esa búsqueda y ese mapped, y el resultado lo tienes que intersectar con lo ya existente. Esa operación de conjuntos lo que hace es dejar sólo los valores que están en ambos conjuntos. Cuidado que había una errata también en la búsqueda, que el template no es lo mismo que el producto cuando hay variantes.

Bueno, como dice el humorista, no lo digo, lo hago:
```
    supplierinfo_obj = self.env['product.supplierinfo']
    # Start with all suppliers
    suppliers = supplier_obj.search([('supplier', '=', True)])
    for line in self.line_ids:
        sinfos = supplierinfo_obj.search(
            [('product_tmpl_id', '=', line.product_id.product_tmpl_id.id)])
        suppliers &= sinfos.mapped('name')
        # Stop condition to avoid the full loop if we don't have suppliers
        if not suppliers:
            break
```